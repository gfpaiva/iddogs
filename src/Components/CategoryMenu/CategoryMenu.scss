@import '../../Scss/root';

.category-menu {
  &__list {
    @include rfs(22);
    @media screen and (max-width: $xs-beakpoint) { @include rfs(16) }
    margin: 0;
    padding: 0;
  }

  &__item {
    display: inline-block;

    &:after {
      @media screen and (max-width: $xs-beakpoint) { margin: 0 .75em }
      content: '/';
      font-weight: 100;
      margin: 0 1em;
    }

    &:last-child:after { display: none }
  }

  &__link {
    font-weight: 100;

    &--active{
      font-weight: 700;
      text-decoration: underline;
    }
  }
}
