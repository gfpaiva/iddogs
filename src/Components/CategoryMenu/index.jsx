import React from 'react';
import { NavLink } from 'react-router-dom';

import Categories from '../../Utils/Categories';

import './CategoryMenu.scss';

export default function CategoryMenu() {
  return (
    <nav className="category-menu text-center my-4">
      <ul className="category-menu__list">
        {Categories.map(({ key, name, category }) => (
          <li className="category-menu__item" key={key}>
            <NavLink
              activeClassName="category-menu__link--active"
              className="category-menu__link"
              to={`/feed/${category}`}
            >
              {name}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
}
