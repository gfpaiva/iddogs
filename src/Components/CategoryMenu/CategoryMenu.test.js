import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import CategoryMenu from './index';

describe('<CategoryMenu />', () => {
  it('should mount with active link properly', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/feed/husky' }]}>
        <CategoryMenu />
      </MemoryRouter>
    );

    expect(wrapper.find('.category-menu')).toMatchSnapshot();
  })
});
