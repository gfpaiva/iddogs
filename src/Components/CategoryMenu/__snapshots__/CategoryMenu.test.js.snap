// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryMenu /> should mount with active link properly 1`] = `
<nav
  className="category-menu text-center my-4"
>
  <ul
    className="category-menu__list"
  >
    <li
      className="category-menu__item"
      key="husky"
    >
      <NavLink
        activeClassName="category-menu__link--active"
        className="category-menu__link"
        to="/feed/husky"
      >
        <Link
          aria-current="page"
          className="category-menu__link category-menu__link--active"
          style={Object {}}
          to="/feed/husky"
        >
          <a
            aria-current="page"
            className="category-menu__link category-menu__link--active"
            href="/feed/husky"
            onClick={[Function]}
            style={Object {}}
          >
            Husky
          </a>
        </Link>
      </NavLink>
    </li>
    <li
      className="category-menu__item"
      key="labrador"
    >
      <NavLink
        activeClassName="category-menu__link--active"
        className="category-menu__link"
        to="/feed/labrador"
      >
        <Link
          aria-current={null}
          className="category-menu__link"
          to="/feed/labrador"
        >
          <a
            aria-current={null}
            className="category-menu__link"
            href="/feed/labrador"
            onClick={[Function]}
          >
            Labrador
          </a>
        </Link>
      </NavLink>
    </li>
    <li
      className="category-menu__item"
      key="hound"
    >
      <NavLink
        activeClassName="category-menu__link--active"
        className="category-menu__link"
        to="/feed/hound"
      >
        <Link
          aria-current={null}
          className="category-menu__link"
          to="/feed/hound"
        >
          <a
            aria-current={null}
            className="category-menu__link"
            href="/feed/hound"
            onClick={[Function]}
          >
            Hound
          </a>
        </Link>
      </NavLink>
    </li>
    <li
      className="category-menu__item"
      key="pug"
    >
      <NavLink
        activeClassName="category-menu__link--active"
        className="category-menu__link"
        to="/feed/pug"
      >
        <Link
          aria-current={null}
          className="category-menu__link"
          to="/feed/pug"
        >
          <a
            aria-current={null}
            className="category-menu__link"
            href="/feed/pug"
            onClick={[Function]}
          >
            Pug
          </a>
        </Link>
      </NavLink>
    </li>
  </ul>
</nav>
`;
